common/src/main/cpp/control/PIDController.cpp

```diff
@@ -46,20 +46,25 @@ void PIDController::SetSetpoint(double setpoint) {
   _setpoint = setpoint;
 }
 
+void PIDController::SetIZone(double threshold) {
+  _threshold = threshold;
+}
+
 double PIDController::GetSetpoint() {
   return _setpoint;
 }
 
 void PIDController::SetWrap(double range) {
-  _wrap_range = range;
+  _wrapRange = range;
 }
 
-double PIDController::Calculate(double processVariable, double dt) {
+double PIDController::Calculate(double processVariable, double dt, double feedforward = 0.0) {
   double error = Wrap(_setpoint - processVariable);
-  _integral += error * dt;
-  _derivative = dt > 0 ? (error - _lastError) / dt : 0;
+  if (std::abs(error) > _threshold) _integral = 0; // I zone
```

Comment:
You need to check if `_threshold` hasn't been set yet! We should make IZone optional, so only if `_threshold > 0` should we do this calculation.

Also, we may need a better name than `_threshold`. Perhaps `_izone_threshold` is a better one